src/mdbExtensionController.ts

```diff
@@ -258,6 +258,30 @@ export default class MDBExtensionController implements vscode.Disposable {
         );
       }
 
+      const disallowedDeepLinkCommands = [
+        EXTENSION_COMMANDS.RUN_PARTICIPANT_CODE,
+        EXTENSION_COMMANDS.OPEN_PARTICIPANT_CODE_IN_PLAYGROUND,
+        EXTENSION_COMMANDS.CONNECT_WITH_PARTICIPANT,
+        EXTENSION_COMMANDS.SELECT_DATABASE_WITH_PARTICIPANT,
+        EXTENSION_COMMANDS.SELECT_COLLECTION_WITH_PARTICIPANT,
+        EXTENSION_COMMANDS.PARTICIPANT_OPEN_RAW_SCHEMA_OUTPUT,
+        EXTENSION_COMMANDS.SEND_MESSAGE_TO_PARTICIPANT,
+        EXTENSION_COMMANDS.SEND_MESSAGE_TO_PARTICIPANT_FROM_INPUT,
+        EXTENSION_COMMANDS.SHOW_EXPORT_TO_LANGUAGE_RESULT,
+        EXTENSION_COMMANDS.MDB_DROP_DATABASE,
+        EXTENSION_COMMANDS.MDB_DROP_COLLECTION,
+        EXTENSION_COMMANDS.MDB_DELETE_DOCUMENT_FROM_TREE_VIEW,
+        EXTENSION_COMMANDS.MDB_DROP_STREAM_PROCESSOR,
+        EXTENSION_COMMANDS.MDB_REMOVE_CONNECTION,
+        EXTENSION_COMMANDS.MDB_REMOVE_CONNECTION_TREE_VIEW,
+      ];
+
+      if (disallowedDeepLinkCommands.includes(command as EXTENSION_COMMANDS)) {
+        throw new Error(
+          `Command '${command}' cannot be invoked via deep links.`,
+        );
+      }
```

Comment:
Non-blocking: Would it make sense to keep a list of allowed commands?

In Compass, we maintain two lists, one of all allowed driver options in connection string and one of all disallowed ones, and through TS we check that they are disjoint and their union is the list of all driver-supported options; that way, if the list of driver options changes, somebody has to modify the lists to make sure we've accounted for it